### DISCUSSION TITLE
Ollama in WSL2, any issues or special things needed to run the UI via docker in Windows?

### DISCUSSION THREAD

**Discussion by ewebgh33** [open-webui/open-webui#510] [Q&A] [answered]
As title says, I have Ollama running fine in WSL2 for Windows11.

I already have docker desktop for Windows 11, and I've tried this out with various LLM apps like Cheshire, Bionic, etc.

I have a problem running Cheshire in Windows Docker, with Ollama in WSL. It doesn't seem to connect to Ollama.

Wondering if I will have a similar problem with the UI.
Can I run the UI via windows Docker, and access Ollama that is running in WSL2?

Would prefer not to also have to run Docker in WSL2 just for this one thing.
WSL2 for Ollama is a stopgap until they release the Windows version being teased (for a year, come onnnnnnn).
So I'm not looking to get heavy into WSL2 specific stuff

Thanks kindly

**Comment by justinh-rahb** [comment]
Hello @EmmaWebGH! Depending on your setup and how you deployed the services, it's possible there are a few factors causing the connectivity problem between Cheshire (or other frontends) and Ollama:

1. Ensure that all the containers (ollama, cheshire, or ollama-webui) reside within the same Docker network.
2. You've deployed each container with the correct port mappings (Example: `11434:11434` for ollama, `3000:8080` for ollama-webui, etc).
3. Use container names as hostnames during container-to-container interactions for proper name resolution, if in doubt you can use `host.docker.internal:11434`.

While using Docker is currently the only supported deployment method, I understand that you and many others would prefer not having to use it to deploy Ollama WebUI or other apps like it. The author has mentioned considering platform-specific installers for future releases: #470

**Comment by ewebgh33** [comment]
Ollama isn't in a docker, it's just installed under WSL2 for windows as I said. So they would not be in a docker network.

Will the Ollama UI, work with a non-docker install of Ollama? As many people are not using the docker version.

I used Autogen Studio and CrewAI today - fresh installs of each. Neither are docker-based. They saw Ollama (also non-docker) just fine.

So I am still unsure about this. Your answer seems to indicate that if Ollama UI and Ollama are both run in docker, I'll be OK. But this is not my case, and also not the case for many Ollama users.

Aside from that, yes everything seems to be on the correct port. Ollama is functioning on the right port, cheshire seems to be functioning on the right port. But that doesn't seem to mean anything in this case as Cheshire can't see Ollama, and so I am wondering if Ollama UI will also not see my Ollama.

Thanks

**Comment by justinh-rahb** [comment]
@EmmaWebGH Ok, thank you for the additional context. You should be able to reach Ollama from Ollama WebUI by specifying the `http://host.docker.internal:11434` as the Ollama API URL. I run the web UI on macOS with Docker Desktop, Ollama is not in my Docker stack, it does work.

**Reply by ewebgh33** [reply]
Thanks for letting me know. I will try this and report back!
Good to know it works that way on Mac. I will see if the same can work on PC. I had not tried that dockerized URL for the API in my CheshireAI tests, maybe that was my issue with that one (and upcoming test with OllamaUI)

**Reply by justinh-rahb** [reply, marked as answer]
Probably that won't help you for Cheshire, it shouldn't have an issue finding Ollama at `http://localhost:11434` (provided you've published port 11434)  as one would normally do. Docker complicates things, as from the container's POV, "localhost" is the container itself and not the host, thus why we need to be more specific in that case.
